src/Components/Nav/Searchbar/index.js

```diff
@@ -0,0 +1,89 @@
+import React, { Component } from 'react';
+import CategoryView from './CategoryView';
+import ResultView from './ResultView';
+import { categoryData } from '../../../Data/categoryData';
+import { characterData } from '../../../Data/characterData';
+import { fetchGet } from '../../../utils/fetches';
+
+export default class Searchbar extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      serachKeyword: '',
+      searchResult: [],
+    };
+  }
+
+  setSerachKeyword = (e) => {
+    this.setState({
+      serachKeyword: e.target.value,
+    });
+  };
+
+  handleReset = () => {
+    this.setState({
+      serachKeyword: '',
+    });
+  };
+
+  handleChangeInput(event) {
+    const searchKeyword = event.target.value;
+    searchKeyword.length === 0 && this.handleReset();
+    fetchGet('', '').then((res) =>
+      this.setState({
+        searchResult: res.result,
+      }),
+    );
+  }
+
+  render() {
+    return (
+      <>
+        <div className="searchModal">
+          <div className="searchForm">
+            <form
+              className="searchInputWrap"
+              onSubmit={() => this.handleReset()}
```

Comment:
```suggestion
              onSubmit={this.handleReset}
```
인자를 넘기지 않을거라면 함수의 인스턴스를 넘기는 것보다 함수 자체를 넘기는 것이 좋습니다. 전자의 경우 더줄여 쓸 수 있는데 더 늘여 쓴 경우이기도 하고, 매 렌더 때마다 새로운 함수를 생성하기 때문에 추후 렌더링 최적화 할 때 처리해줄 필요가 생길 수 있겠습니다.

Reply:
수정하였습니다